macOS/DuckDuckGo/Tab/ViewModel/TabViewModel.swift

```diff
@@ -96,7 +96,26 @@ final class TabViewModel {
     }
 
     var canPrint: Bool {
-        !isShowingErrorPage && canReload && tab.webView.canPrint
+        guard !isShowingErrorPage else { return false }
+        switch tab.content {
+        case .url(let url, _, _):
+            return !(url.isDuckPlayer || url.isDuckURLScheme) && canReload && tab.webView.canPrint
+        case .history:
+            return false
+        default:
+            return canReload && tab.webView.canPrint
+        }
+    }
+
+    var canShare: Bool {
+        switch tab.content {
+        case .url(let url, _, _):
+            return !(url.isDuckPlayer || url.isDuckURLScheme)
```

Comment:
❓ I’m confused about this one. Why share is disabled for Duck Player?

Reply:
This was the decision when we released Duck Player for the first time. We don’t encourage sharing duck URLs (because our browsers can’t open Duck URLs from outside of the app, at the moment) and Duck Player UI (or the actual YouTube iframe) provides a “copy video URL” button that uses the actual youtube video URLs. We could revisit it any time, but I didn’t want to change this behavior with my patch.

Reply:
Thanks for the explanation! This is good to go :shipit: